osu.Game.Rulesets.Osu/Difficulty/Skills/Speed.cs

```diff
@@ -42,7 +44,21 @@ protected override double StrainValueOf(DifficultyHitObject current)
             var osuCurrent = (OsuDifficultyHitObject)current;
 
             double distance = Math.Min(single_spacing_threshold, osuCurrent.TravelDistance + osuCurrent.JumpDistance);
-            double deltaTime = Math.Max(max_speed_bonus, current.DeltaTime);
+            double deltaTime = current.DeltaTime;
+
+            // Aim to nerf cheesy rhythms (Very fast consecutive doubles with large deltatimes between)
+            double deltaTimeThreshold = greatWindow * 2;
+
+            if (Previous.Count > 0 && deltaTime < deltaTimeThreshold && Previous[0].DeltaTime > deltaTime)
+            {
+                double speedWindowRatio = deltaTime / deltaTimeThreshold;
+                deltaTime = Interpolation.Lerp(Previous[0].DeltaTime, deltaTime, speedWindowRatio);
```

Comment:
Let me confirm the interpretation of this line: The closer the delta time is to 0, the harsher this nerf is (because the resulting value of the lerp is closer to `Previous[0].DeltaTime` which is the larger one). And the closer it is to `deltaTimeThreshold`, the more lenient the nerf is. Is that correct?

I can see that making sense in a way but I'd like to confirm my understanding of this to be sure as it may seem a little counterintuitive at first.

Reply:
That is correct.

Reply:
Yeah, that's correct. For some added detail, this was previously a harsh cap but implementing a lerp was necessary since values close to `deltaTimeThreshold` (which are very unlikely to be cheeseable doubles) were getting the full nerf.